### ISSUE TITLE
node process uses 100% of CPU

### ISSUE THREAD

**Issue by kilianc** [SublimeLinter/SublimeLinter-jshint#70]
it happens consistenly with node v0.12.2

**Comment by acburdine** [comment]
I'm experiencing this as well.

**Comment by kilianc** [comment]
I had to remove the package, it makes sublime unusable.

**Comment by alexblunck** [comment]
Same here

**Comment by jeduan** [comment]
Same here

**Comment by anotheri** [comment]
:+1:

**Comment by niftylettuce** [comment]
:+1:

**Comment by alexblunck** [comment]
Not 100% sure why, but simply re-installing jshint via npm seems to have fixed the issue for me.

Haven't seen the 100% CPU issue in the last 3 days.

Here's the command I used:

`$ sudo npm install -g jshint`

**Comment by kilianc** [comment]
@alexblunck it will be back don't worry, tired multiple times stays away for a while then kicks back in.

**Comment by alexblunck** [comment]
@kilianc You're right, it's back.

**Comment by holm** [comment]
This is happening for me also with io.js 2.0.2

**Comment by wolfbiter** [comment]
+1

**Comment by sedatkimya** [comment]
+1 CPU usage %100

**Comment by oitozero** [comment]
+1

**Comment by kbjr** [comment]
It actually seems like the node process sticks around after closing sublime. So if you close sublime and reopen it, you now have 2 node processes, etc. And once you start going down that route the cpu usage grows quickly.

**Comment by kilianc** [comment]
@kbjr if you are saying that the reason why we see high CPU usage is because the processes stack up on each sublime run, that is not correct.

Every node process spawned by sublime takes 100% CPU, I recall killing a couple of node processes manually but I am not sure if orphan node processes are just a side effect of node hanging or happens systematically.

Either way this needs to be fixed ASAP.

**Comment by kbjr** [comment]
@kilianc When I looked at it, I had 4 separate node processes taking 25% each. Killing each one helped, no one process seems able to take 100% for me.

I do agree though. Regardless, this needs a fix.

**Comment by mokkabonna** [comment]
If we are to look into this, we need to find a common factor. So far people are just adding +1's What we need is for people to at least give their versions of node, jshint and sublimelinter-jshint.

Any other related info is also welcome, like does it happen all the time, or only after it encounters some specific error, or after a specific duration etc etc.

I personally are not seeing any of this with node 0.10.25 and jshint 2.7.0 and latest version of this plugin.

**Comment by kilianc** [comment]
@mokkabonna `v0.10.25` has been released on 01/23/2014 -_- not really a good reference for debugging. Is that the last supported version?

I am on latest node `v0.12.4` stable on mac.
No specific error, just crazy CPU.

**Comment by mokkabonna** [comment]
This plugin really only have requirements of jshint 2.5.0 or later. Other than that it should work with any node really, meaning any node that can run 2.5.0 of jshint.

So 0.10.25 is as good as any really.

**Comment by kilianc** [comment]
jshint `v2.7.0` here

**Comment by acburdine** [comment]
I have node v0.12.2 on mac, running jshint v2.7.0 and latest version of the plugin and I'm getting the issue.

**Comment by holm** [comment]
I have io.js v2.0.2 and jshint v2.7.0. I have had no luck finding out how to consistently reproduce this. When it happens there is a iojs/node process running with 100% CPU and the highlighting stops working. Killing the process will make the highlighting work again on subsequent saves. 

My best guess is that it happens when you rapidly save the file, so a background lint is running during the save, but that is just a guess.

**Comment by kbjr** [comment]
I'm running node@0.12.0 and jshint@2.7.0

**Comment by Saartje87** [comment]
+1 :)
Hope this could help a bit :) Not sure when is crashes tho.. Randomly my mac start making helicopter sound.

node v0.12.4 and jshint v2.7.0.

ps aux | grep node

User          8055  99.7  0.2  3067572  35040   ??  R     3:42PM 547:26.62 node /usr/local/bin/jshint --verbose --filename /path/to/file/login_ctrl.js --config /path/to/file/.jshintrc

**Comment by mokkabonna** [comment]
Have anyone experienced this with other node based linters or just jshint?

**Comment by emphaticpuma** [comment]
Exact same issue on Ubuntu 15.04.

**Comment by scottwrobinson** [comment]
OSX v10.9.5
jshint v2.5.11
sublimelinter-jshint: v1.2.1
io.js 2.2.1

Seems to happen after a few hours of using Sublime.

**Comment by scottwrobinson** [comment]
Ok, bear with me here, but I think I found out how to recreate this...

I was working on [this](https://gist.github.com/scottwrobinson/e7f34b9fced0a1c36b05) code today for my [Camo](https://github.com/scottwrobinson/camo) project and had Activity Monitor running to try and catch when this happens. I ended up having to retype the code from [line 484](https://gist.github.com/scottwrobinson/e7f34b9fced0a1c36b05#file-sublimelinter-jshint-issue-70-js-L484) multiple times and noticed that's when `iojs` would take up 100% of the CPU.

(Sorry, I tried to trim down the file but then I couldn't recreate the problem.)

Just follow the directions described by comments on lines [482](https://gist.github.com/scottwrobinson/e7f34b9fced0a1c36b05#file-sublimelinter-jshint-issue-70-js-L482) and [489](https://gist.github.com/scottwrobinson/e7f34b9fced0a1c36b05#file-sublimelinter-jshint-issue-70-js-L489) to do it. It seems like it happens more often when I let Sublime's autocomplete finish the longer words for me. Also, doesn't seem to happen until I type in the last semicolon on line [485](https://gist.github.com/scottwrobinson/e7f34b9fced0a1c36b05#file-sublimelinter-jshint-issue-70-js-L485).

Hope this helps. Let me know if you have any questions.

**Comment by CavemanNinja** [comment]
+1
node v0.10.25

jshint v2.7.0

Ubuntu 15.04

**Comment by felixdulfer** [comment]
+1 
node v4.0.0 (installed via nvm)
jshint v2.8.0
os x 10.10.5

I'm also having this issue when I'm writing the `class` keyword.

**Comment by flockonus** [comment]
+1
node v4.1.0 & v4.2.1 (via nvm)
jshint v2.8.0
os x 10.10.5

I'm also having this issue when I'm writing the `class` keyword. (2)

**Comment by SamFromDaUk** [comment]
+1

**Comment by versedi** [comment]
Constant 25% CPU usage per core. +1

**Comment by kilianc** [comment]
main reason why I switched to atom TBH

**Comment by qqilihq** [comment]
Had the same issues and just found this ticket. I just updated to `jshint@2.9.1-rc3` and the problem seems to be gone, at least for the `class` example given above. I'll keep an eye on it with our own codebase during the next day.s

**Comment by gueletk** [comment]
I'm using Atom on Ubuntu 14.04 and I just noticed this issue as well. node v4.2.6

**Comment by adametry** [comment]
About once a day, I'll notice my laptop fans speed up and, like a notification, I open Activity Monitor to find & kill a "node" process floating around 100% CPU usage. If this happens while I still have Sublime Text open, I can see that JSHint is no longer provides feedback. Also, The process will stick around after ST is closed. I find this often happens after frequently switching tabs or writing new JS files; however, it's rare to notice it immediately.

This time, I found the following error in ST's console (hopefully, not a red-herring):
```
SublimeLinter: error in SublimeLinter daemon: 
SublimeLinter: -------------------- 
SublimeLinter: Traceback (most recent call last):
  File "~/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/queue.py", line 57, in loop
    item = self.q.get(block=True, timeout=self.MIN_DELAY)
  File "./queue.py", line 175, in get
queue.Empty

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "~/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/queue.py", line 65, in loop
    self.lint(view_id, timestamp)
  File "~/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/queue.py", line 111, in lint
    self.callback(view_id, timestamp)
  File "~/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/sublimelinter.py", line 119, in lint
    Linter.lint_view(view, filename, code, hit_time, callback)
  File "~/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/linter.py", line 957, in lint_view
    callback(cls.get_view(vid), linters, hit_time)
  File "~/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/sublimelinter.py", line 137, in highlight
    vid = view.id()
AttributeError: 'NoneType' object has no attribute 'id'
 
SublimeLinter: -------------------- 
```

I have the following relevant software versions:
OS X (10.11.3)
Sublime Text 3 (buid 3102)
SublimeLinter (3.6.1)
SublimeLinter-jshint (1.2.1)
jshint (2.9.1)
node (5.5.0 via nvm)

**Comment by nchery** [comment]
+1
OS X(10.10.4)
SublimeText 3
Jshint v2.7.0
node v4.2.1

any workaround by chance? thinking of switching to another IDE

**Comment by DiegoZoracKy** [comment]
Same...

OS X (10.11.6)
SublimeText 3 (build 3114)
Jshint v2.9.3
node v6.5.0

**Comment by markyun** [comment]
+1
OS X(10.11.5)
SublimeText 3
Jshint v2.7.0
node v4.6.0

**Comment by iarobinson** [comment]
I was doing a project Euler using:

Atom 1.11.2
maxOS Sierra - Version 10.12
node 4.4.7

After running some high range computational programs a few times, I had 3 instances of node using 100% of my CPU each. **To stop the processes, I did the following**:

- Open Activity Monitor
- Double click 'node' which is causing problem
- Click 'Quit' on the pop-up window that shows up.

After about 60 seconds my computer stopped being in a state of overdrive.

**Comment by rwallace-kabam** [comment]
Did anyone ever figure this out?  The problem still occurs on jshint 2.9.5 and the latest sublime / sublimelinter-jshint

SublimeLinter 4.2.1
SublimeLinter-jshint 1.3.4

It makes this linter basically un-usable as my machine will spin up 20+ node processes and completely consume all available cpu

**Comment by kaste** [comment]
Fwiw: Next or next-next release will actively send kill/terminate signals to old processes.

**Comment by braver** [comment]
This "next next" release it out!

**Comment by chandragithub** [comment]
Solved: I switched to node version 12.6.0 to 10.16.0 and the problem got resolved for me.
